projects/storefrontapp-e2e-cypress/cypress/integration/regression/infinite-scroll.e2e-spec.ts

```diff
@@ -1,50 +1,84 @@
 import {
-  infiniteScrollNoShowMoreTest,
-  infiniteScrollNotActivatedTest,
-  infiniteScrollWithShowMoreAtALimit,
-  infiniteScrollWithShowMoreAtTheBeginningTest,
-  scrollConfig,
-  verifyProductListLoaded,
+  configScroll,
+  isPaginationNotVisible,
+  backtoTopIsNotVisible,
+  scrollToFooter,
+  backToTopIsVisible,
+  isPaginationVisible,
+  verifySortingResetsList,
+  verifyFilterResetsList,
+  verifyGridResetsList,
 } from '../../helpers/infinite-scroll';
+import {
+  createAllProductQuery,
+  QUERY_ALIAS,
+} from '../../helpers/product-search';
+
+describe('Infinite scroll', () => {
+  const testUrl = '/Open-Catalogue/Components/Power-Supplies/c/816';
+  const productQueryAlias = `@${QUERY_ALIAS.INFINITE_SCROLL_PRODUCT_LOADED}`;
 
-context('Infinite scroll', () => {
   before(() => {
     cy.window().then((win) => win.sessionStorage.clear());
   });
 
-  describe('infinite scroll is active and no show more button', () => {
-    beforeEach(() => {
-      scrollConfig(true, 0, false);
-      verifyProductListLoaded();
-    });
-
-    infiniteScrollNoShowMoreTest();
+  beforeEach(() => {
+    cy.server();
+    createAllProductQuery(QUERY_ALIAS.INFINITE_SCROLL_PRODUCT_LOADED);
   });
 
-  describe('infinite scroll should be active using the show more button from the beginning', () => {
-    beforeEach(() => {
-      scrollConfig(true, 0, true);
-      verifyProductListLoaded();
-    });
+  it("should enable Infinite scroll and NOT display 'Show more' button", () => {
+    configScroll(true, 0, false);
+    cy.visit(testUrl);
 
-    infiniteScrollWithShowMoreAtTheBeginningTest();
-  });
+    cy.wait(productQueryAlias).then((waitXHR) => {
+      const totalResults = waitXHR.response.body.pagination.totalResults;
+      isPaginationNotVisible();
+
+      backtoTopIsNotVisible();
+      scrollToFooter(totalResults);
+      backToTopIsVisible();
+
+      verifySortingResetsList();
+
+      verifyFilterResetsList();
```

Comment:
this makes the test fail always in this way of writing the e2e in one `it` 

I believe you would need to target the specific url for sorting / filterring / gridreset - search endpoint

If it runs only have 1 of them, the test passes, but if we have two or more assertions assertions to that route, then it says the screenshot below.

tested on 2005 server, develop, and dev-6 in both the testrunner and headless version

<img width="443" alt="Screen Shot 2020-07-08 at 5 22 46 PM" src="https://user-images.githubusercontent.com/29099466/86971736-ac894880-c13f-11ea-833a-87aaa9a9c106.png">

Reply:
hmm tried it again, and it passed, I would say it's flaky 
<img width="404" alt="Screen Shot 2020-07-08 at 5 38 00 PM" src="https://user-images.githubusercontent.com/29099466/86973080-f70bc480-c141-11ea-8477-7590f533e713.png">
<img width="398" alt="Screen Shot 2020-07-08 at 5 38 10 PM" src="https://user-images.githubusercontent.com/29099466/86973153-1c98ce00-c142-11ea-868b-7c1059c3cd21.png">

Reply:
jenkins build #jenkins-394 https://jkmaster.test.c3po.b2c.ydev.hybris.com/job/spartacus-regression-tests-for-branch/394/consoleFull shows a new error that I never seen running it

jenkins build #jenkins-395-396 displays the same result I found above https://jkmaster.test.c3po.b2c.ydev.hybris.com/job/spartacus-regression-tests-for-branch/395/consoleFull
https://jkmaster.test.c3po.b2c.ydev.hybris.com/job/spartacus-regression-tests-for-branch/396/consoleFull

Reply:
Great findings Brian, thank you. Addressed now.